invenio_edugain/utils.py

```diff
@@ -57,6 +62,65 @@ def get_idp_data_dict() -> dict:
     }
 
 
+def load_mdstore(
+    metadata_xml_location: PathLike | str,
+    cert_location: PathLike | str | None = None,
+    fingerprint_sha256: str | None = None,
+    *,
+    http_client_timeout: int = 30,
+) -> MetadataStore:
+    """Load a pysaml2 MetadataStore from given path/url.
+
+    When loading metadata from url, requires a certificate to check validity of metadata.
+    When loading that certificate from url, requires a fingerprint to check validity of certificate.
+    """
+    mds = MetadataStore(None, Config(), http_client_timeout=http_client_timeout)
+
+    if isinstance(metadata_xml_location, str) and validators.url(metadata_xml_location):
+        # load metadata from url: need cert, either via local file or remote fingerprint-checked cert
+        if cert_location is None:
+            msg = "must provide a certificate when loading metadata-xml from URL"
+            raise TypeError(msg)
+
+        if isinstance(cert_location, str) and validators.url(cert_location):
+            # load cert from url: download cert and check fingerprint
+            if fingerprint_sha256 is None:
+                msg = "must provide a fingerprint when loading certificate from URL"
+                raise TypeError(msg)
```

Comment:
this function runs in a celery task or?
as i see the raised exceptions are not trapped somewhere.
to see the errors on the invenio-jobs log feature it would be necessary to write them to the `current_app.logger.error` functionality

Reply:
ok, yes, with manually trapping and using `current_app.logger.error` you would not see the stack trace. so the error message would be "nicer" . we could discuss if the stack trace would give some valuable insights, but normally i think it is not necessary and a one line error message would be enough, since the error message is enough to locate the error and the error message should be clear enough anyway to know what should be done